.github/workflows/ci.yml

```diff
@@ -0,0 +1,156 @@
+        command: test
+        args: --release --target ${{ matrix.target }} --bin rustup-init
+
+    - name: tests - lib
+      uses: actions-rs/cargo@v1
+      with:
+        command: test
+        args: --release --target ${{ matrix.target }} --lib --all
+
+    - name: tests - doc
+      uses: actions-rs/cargo@v1
+      with:
+        command: test
+        args: --release --target ${{ matrix.target }} --doc --all
+
+    - name: tests - cli-exact
+      uses: actions-rs/cargo@v1
+      with:
+        command: test
+        args: --release --target ${{ matrix.target }} --test cli-exact
+
+    - name: tests - cli-inst-interactive
+      uses: actions-rs/cargo@v1
+      with:
+        command: test
+        args: --release --target ${{ matrix.target }} --test cli-inst-interactive
+
+    - name: tests - cli-misc
+      uses: actions-rs/cargo@v1
+      with:
+        command: test
+        args: --release --target ${{ matrix.target }} --test cli-misc
+
+    - name: tests - cli-rustup
+      uses: actions-rs/cargo@v1
+      with:
+        command: test
+        args: --release --target ${{ matrix.target }} --test cli-rustup
+
+    - name: tests - cli-self-upd
+      uses: actions-rs/cargo@v1
+      with:
+        command: test
+        args: --release --target ${{ matrix.target }} --test cli-self-upd
+
+    - name: tests - cli-v1
+      uses: actions-rs/cargo@v1
+      with:
+        command: test
+        args: --release --target ${{ matrix.target }} --test cli-v1
+
+    - name: tests - cli-v2
+      uses: actions-rs/cargo@v1
+      with:
+        command: test
+        args: --release --target ${{ matrix.target }} --test cli-v2
+
+    - name: tests - dist_install
+      uses: actions-rs/cargo@v1
+      with:
+        command: test
+        args: --release --target ${{ matrix.target }} --test dist_install
+
+    - name: tests - dist_manifest
+      uses: actions-rs/cargo@v1
+      with:
+        command: test
+        args: --release --target ${{ matrix.target }} --test dist_manifest
+
+    - name: tests - dist
+      uses: actions-rs/cargo@v1
+      with:
+        command: test
+        args: --release --target ${{ matrix.target }} --test dist -- --test-threads 1
+
+    - name: tests - dist_transactions
+      uses: actions-rs/cargo@v1
+      with:
+        command: test
+        args: --release --target ${{ matrix.target }} --test dist_transactions
+
+  check_fmt:
+    name: Checking rustfmt
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@master
+
+    - uses: actions-rs/toolchain@v1
+      with:
+          toolchain: beta
+          components: rustfmt
+          override: true
+          profile: minimal
+
+    - name: fmt
+      run: cargo fmt --all -- --check
+
+  check_clippy:
+    name: Checking clippy
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v1
+      - run: rustup component add clippy
+      - uses: actions-rs/clippy-check@v1
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
```

Comment:
I don't familiar with actions-rs. Why does it need GITHUB_TOKEN?

Reply:
I think that lets it post results back directly onto the PR